src/core/ddsc/src/dds_whc.c

```diff
@@ -403,7 +464,7 @@ struct whc *whc_new (struct q_globals *gv, int is_transient_local, uint32_t hdep
   whc->seq_hash = ddsrt_hh_new (1, whc_node_hash, whc_node_eq);
 #endif
 
-  if (whc->idxdepth > 0)
+  if (wrinfo->idxdepth > 0)
```

Comment:
Not having an index for a KEEP_ALL topic is attractive for performance, but the downside is that there is then no instance at all to use in a deadline administration. Correctly implementing register/unregister appears to require one as well, so it appears one should be created always.

So as things stand now, I suspect "deadline" won't work for a KEEP_ALL volatile writer and for a KEEP_ALL transient-local writer with a KEEP_ALL durability_service.history setting.